### DISCUSSION TITLE
wb$open() failing to open `libreoffice` (not openxlsx2 error)

### DISCUSSION THREAD

**Discussion by J-Moravec** [JanMarvin/openxlsx2#1174] [Q&A] [answered]
**tl,dr:** `system2("soffice")` cannot find libre office libraries, but `soffice` works from terminal.

Freshly reinstalled `openxlsx2`, libreoffice works from terminal with both `soffice` and `libreoffice` (Ubuntu 22.04 LTS "Jammy")

When trying to run an `openxlsx2` example with `wb$open()`, it fails with:
```
wb = wb_workbook()
wb$open()

# /usr/lib/libreoffice/program/javaldx: error while loading shared libraries: libreglo.so: cannot open shared object file: No such file or directory
# Warning: failed to read path from javaldx
# /usr/lib/libreoffice/program/soffice.bin: error while loading shared libraries: libreglo.so: cannot open shared object file: No such file or directory
```

Since `libreoffice` works from the command line, this has to do something with running the binary from R,
since when drilling down, `openxlsx2`'s `wb$open()` just runs `system2("usr/bin/soffice")` (in my case).
And running it this way reproduces the error:

```
system2("soffice")
# /usr/lib/libreoffice/program/javaldx: error while loading shared libraries: libreglo.so: cannot open shared object file: No such file or directory
# Warning: failed to read path from javaldx
# /usr/lib/libreoffice/program/soffice.bin: error while loading shared libraries: libreglo.so: cannot open shared object file: No such file or directory
# Warning message:
# In system2("/usr/bin/soffice") : error in running command

```

`openxlsx2`  and ubuntu version:
```
packageVersion("openxlsx2")
# 1.10
```

```
lsb_release -a
# Distributor ID:	Ubuntu
# Description:	Ubuntu 22.04.5 LTS
# Release:	22.04
# Codename:	jammy
```

Help is greatly appreciated.

**Comment by JanMarvin** [comment]
Hi @J-Moravec , I haven’t used a Debian based desktop Linux in ages, but it looks like whatever is found in /usr/bin/soffice is missing libreglo.so. But this should trigger an error if you run /usr/bin/soffice in the terminal. Maybe your system is missing libreglo-dev?

**Comment by JanMarvin** [comment]
Do you have another soffice installation as well? Maybe via flatpak or snap?

**Comment by JanMarvin** [comment]
Or maybe this is the issue: https://askubuntu.com/a/1495121

not sure how R handles environment variables like LD_LIBRARY_PATH

**Comment by J-Moravec** [comment]
See above, I can run libreoffice just fine, from GUI or command line. Both through `soffice` and `libreoffice`.

`libreglo.so` is present in its cozy place at `/usr/lib/libreoffice/program/libreglo.so`.

I can just save it and open it manually, so it is not that of a big deal, I just hoped that someone had a similar problem in past.

**Comment by JanMarvin** [comment, marked as answer]
See a similar issue with an older Ubuntu version: https://stackoverflow.com/a/42044762

In your R environment the shared library is not found. It should work if you run R in the console where you can open soffice. But the shared library is not found from wherever you tried to run R in the opening post. Maybe an issue with your ide or some misconfiguration. Either way, I move this to the discussion section as it’s not an openxlsx2 issue.

**Comment by J-Moravec** [comment]
Thanks. Couldn't find the Dirk's answer.

Yes, something was weird with the library linkage, but couldn't figure out what exactly.
